appmap/test/test_pytest.py

```diff
@@ -0,0 +1,37 @@
+import os
+import os.path
+import subprocess
+
+import json
+
+from .appmap_test_base import AppMapTestBase
+
+
+class TestPytest(AppMapTestBase):
+    def compose_exec(self, args, test_dir, tmpdir):
+        cwd = test_dir
+        env = {
+            "PATH": os.getenv("PATH"),
+            "APPMAP_OUTPUT_DIR": tmpdir
+        }
+        return subprocess.run(['docker-compose'] + args,
```

Comment:
Would it be possible to not use docker here? Using docker here is quite unexpected. I don't have an appropriate python available in my main environment and tried to run the tests in docker. This failed utterly. (Of course it's possible to make it work, but it's fiddly and brittle.)

It seems to work to pretty much just call `poetry run pytest` in the correct directory instead of docker-compose. A separate dockerfile and script can be provided to run the whole thing in docker for convenience.

Reply:
The most important thing here is to ensure that installing `appmap-python` into a project that uses `pytest` configures everything correctly. Using docker is an easy way to do that. Both `appmap-ruby` and `appmap-java` have some docker-based testing for basically the same purpose. 

There is a Dockerfile and an entrypoint script as part of this PR. It should be easy enough to get rid of the use docker-compose, but I think we should keep using docker for the test.

Reply:
@apotterri have you had a chance to look at this: https://docs.pytest.org/en/stable/writing_plugins.html#testing-plugins

Reply:
Just took a look. It's very specific to pytest, of course. I still feel like a docker-based approach gives us the easiest way to get a clean environment, and will be applicable to test all the different integrations.

Reply:
The problem with using docker is it makes it quite difficult to run it inside docker... at the very least the docker-based tests should be optional and easily skipped. (It has always annoyed me in `appmap-ruby`, too.)

Reply:
I believe [tox](https://tox.readthedocs.io/) is a tool that could help setting up the environment to test it all sans docker.

Reply:
I looked at tox and at pytest's support for testing plugins. The latter looked like an easier way to get up and running quickly, so I went with that.